crates/editor/src/editor.rs

```diff
@@ -6153,6 +6153,34 @@ impl Editor {
         }
     }
 
+    pub fn reload_file(&mut self, _: &ReloadFile, cx: &mut ViewContext<Self>) {
+        // directly copied this because i don't know how to call it :P
+        // TODO: call reload from crates/editor/src/items.rs:788 instead of copying it here
+        if let Some(project) = self.project.clone() {
+            let buffer = self.buffer().clone();
+            let buffers = self.buffer.read(cx).all_buffers();
+            let reload_buffers =
+                project.update(cx, |project, cx| project.reload_buffers(buffers, true, cx));
+
+            // hacky to assign _ here
+            let _ = cx.spawn(|this, mut cx| async move {
+                let transaction = reload_buffers.log_err().await;
+                this.update(&mut cx, |editor, cx| {
+                    editor.request_autoscroll(Autoscroll::fit(), cx)
+                }).ok();
+                buffer
+                    .update(&mut cx, |buffer, cx| {
+                        if let Some(transaction) = transaction {
+                            if !buffer.is_singleton() {
+                                buffer.push_transaction(&transaction.0, cx);
+                            }
+                        }
+                    })
+                    .ok();
+            });
+        }
+    }
```

Comment:
i'm going to attempt to figure this out but if someone happens to look at this and knows what i need to do to call the [existing reload function](https://github.com/schpet/zed/blob/b5215fc7eceb3e27b9e09221d533dba0a553981c/crates/editor/src/items.rs#L788) i'm interested in learning (i don't know rust very well at all)

Reply:
You should just be able to call `self.reload()` in your action callback. Does that not work?

Reply:
oh! thanks. https://github.com/zed-industries/zed/pull/18395/commits/c3a72e18b3ba1c4ba4b21782432d1485e90a86a4 attempts to do that. i am curious if

- is `Item as WorkspaceItem` legit? 
- is `let _ = self.reload(project, cx);` legit or should i try to do something with the result

Reply:
oh: also wondering if project.clone() makes sense

Reply:
You probably need the clone to keep the borrow checker happy (`=` may not do what you might expect in rust https://cirw.in/blog/rust-assignment).

With the result you should be able to call `.notify_err()` on the result to have it show up:

```rust
 pub fn reload_file(&mut self, _: &ReloadFile, cx: &mut ViewContext<Self>) {
        let Some(workspace) = self.workspace() else {
           return;
        }
        let Some(project) = self.project.clone() else {
            return;
        }
        self.reload(project, cx).notify_err(workspace, cx)
    }
```

(though I haven't tried running it).

Reply:
huh! really appreciate the link and the post. 

landed on 1b1a690 - `self.reload(project, cx).detach_and_notify_err(cx);`